_posts/2023-8-22-introducing-sqlite3-to-kw.md

```diff
@@ -0,0 +1,114 @@
+
+For statistics, a file `statistics/<year>/<month>/<day>` represented statistics collected at _\<month\>/\<day\>/\<year\>_. For example, a line
+
+```
+build 497
+```
+
+in a file `statistics/23/08/23`, meant that a `kw build` command ran on August 23 of 2023 and lasted for 8 minutes and 17 seconds (497 seconds).
+
+`kw pomodoro` had this same file structure that represented dates, with each line representing an entry. Differently from the statistics database though, each line/entry was comma separated, had a different number of attributes, and also had an optional attribute. On top of that, there was a file `~/local/share/kw/pomodoro/tags` for storing Pomodoro tags and a file `~/local/share/kw/pomodoro_current.log` for the active Pomodoro timeboxes.
+
+I could also explain the intricacies of the `kw kernel-config-manager` database (which had even more particularities), but it would probably be tiresome for you the reader.
+
+The point may be already clear: although functional, each feature had to implement its own database with its own details. This made the code hard to scale and more coupled with these particularities of **where** and **how** the data was stored.
+
+## The right DBMS
+
+DBMSs are really vast and diverse, proposing different solutions for different problems. The people involved in kw knew that the introduction of a DBMS was necessary and agreed on some requisites for the system:
+
+- Be _Free Libre and Open Source Software_ (FLOSS).
+- Have a CLI interface for easy integration with Bash script, as we want to maintain kw's codebase in pure bash wherever possible.
+- Have a small footprint.
+- Run on user space.
+- Be a Relational DBMS.
+
+In the end, the DBMS that was chosen was SQLite3, as it was Public Domain (not exactly FLOSS, but **much** better than proprietary), had a CLI interface, sized less than 1 MB, ran on user space, and was Relational.
+
+## kw new database
+
+> The following description is based on the `unstable` branch at commit `#02e89e2`, which was the last commit of the [PR](https://github.com/kworkflow/kworkflow/pull/836) that introduced SQLite3 to kw. You can check kw's repo at this state [here](https://github.com/kworkflow/kworkflow/tree/02e89e22983528573fa968516533c18b0de8c12e).
+
+First, I must point out that kw's database schema, with all the tables, views, indexes, and triggers was a wonderful job made by [Rubens Gomes Neto](https://github.com/kwy95) and [Magali Lemes](https://github.com/magalilemes) and is described at `database/kwdb.sql`.
```

Comment:
It would be nice if you add a very brief introduction to our DB schema. Take the diagram from Ruben's capstone project, add it here, and explain just a little bit about that; next point it to @kwy95 final text.